test/boost/reader_concurrency_semaphore_test.cc

```diff
@@ -1156,14 +1156,14 @@ SEASTAR_THREAD_TEST_CASE(test_reader_concurrency_semaphore_evict_inactive_reads_
     BOOST_REQUIRE_EQUAL(semaphore.available_resources().count, 0);
     BOOST_REQUIRE(irh1);
 
-    // Marking p2 as awaits should now allow p3 to be admitted by evicting p1
+    // Marking p2 as awaits should eventually allow p3 to be admitted by evicting p1
     rd2.mark_as_awaits();
-    BOOST_REQUIRE_EQUAL(semaphore.get_stats().waiters, 0);
-    BOOST_REQUIRE_EQUAL(semaphore.get_stats().need_cpu_permits, 1);
-    BOOST_REQUIRE_EQUAL(semaphore.get_stats().awaits_permits, 1);
-    BOOST_REQUIRE_EQUAL(semaphore.get_stats().inactive_reads, 0);
-    BOOST_REQUIRE_EQUAL(semaphore.get_stats().permit_based_evictions, 1);
-    BOOST_REQUIRE_EQUAL(semaphore.available_resources().count, 0);
+    REQUIRE_EVENTUALLY_EQUAL(semaphore.get_stats().waiters, 0);
+    REQUIRE_EVENTUALLY_EQUAL(semaphore.get_stats().need_cpu_permits, 1);
```

Comment:
I think only the first needs to be `_EVENTUALLY`, once `maybe_admit_readers()` is scheduled, all the other checks should pass immediately.

Reply:
True, but wanted to emphasize it in case the checks get shuffled. It is also more readable, I think.

Reply:
The checks cannot shuffle. As-is the code gives you the impression, that there can be a delay between the different checks becoming "green", whereas what really happen is that we are waiting for an event, which should make all checks pass atomically.

Reply:
Fixed